app/models/manageiq/providers/amazon/inventory/collector/target_collection.rb

```diff
@@ -12,7 +13,8 @@ def references(collection)
   end
 
   def instances
-    hash_collection.new(
+    return @instances unless @instances.blank?
```

Comment:
? :) !

Yeah, any? present?. I'm just saying `return if` is easier to grok than `return unless`. 

Always try to use a positive version...